workflow/controller/indexes/labels.go

```diff
@@ -27,6 +28,20 @@ func MetaWorkflowPhaseIndexFunc() cache.IndexFunc {
 	}
 }
 
+func MetaPodPhaseIndexFunc() cache.IndexFunc {
+	return func(obj interface{}) ([]string, error) {
+		pod, ok := obj.(*v1.Pod)
+
+		if !ok {
+			return nil, nil
+		}
+		if pod.Status.Phase == "" {
```

Comment:
I don't think we should have this check - empty string appearing in phases would be fine

Reply:
I added this because there was a difference between `[]string{""}` and `[]string{}`, not sure this difference is relevant anymore. Will remove if possible